crates/sui-cost/src/estimator.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022, Mys&ten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs;
+use std::str::FromStr;
+use strum_macros::Display;
+use strum_macros::EnumString;
+use sui_types::gas::GasCostSummary;
+use sui_types::messages::SingleTransactionKind;
+use sui_types::messages::TransactionKind;
+
+pub const ESTIMATE_FILE: &str = "data/estimate.toml";
+#[derive(
+    Debug, Eq, PartialEq, Hash, Serialize, Deserialize, Ord, PartialOrd, Clone, Display, EnumString,
+)]
+pub enum CommonTransactionCosts {
+    Publish,
+    MergeCoin,
+    SplitCoin(usize),
+    TransferWholeCoin,
+    TransferWholeSuiCoin,
+    TransferPortionSuiCoin,
+    SharedCounterCreate,
+    SharedCounterAssertValue,
+    SharedCounterIncrement,
+}
+
+pub fn estimate_computational_costs_for_transaction(
+    tx_kind: TransactionKind,
+) -> Result<GasCostSummary, anyhow::Error> {
+    let cost_map = read_estimate_file()?;
+    let unsupported_tx_kind = Err(anyhow!("Transaction kind not supported for estimator yet"));
+    match tx_kind {
+        TransactionKind::Single(s) => match s {
+            SingleTransactionKind::TransferSui(t) => Ok(if t.amount.is_none() {
+                cost_map.get(&CommonTransactionCosts::TransferWholeSuiCoin)
+            } else {
+                cost_map.get(&CommonTransactionCosts::TransferPortionSuiCoin)
+            }
+            .unwrap()
+            .clone()),
+
+            SingleTransactionKind::TransferObject(_) => unsupported_tx_kind,
+            SingleTransactionKind::Publish(_) => unsupported_tx_kind,
+            SingleTransactionKind::Call(_) => unsupported_tx_kind,
+            SingleTransactionKind::ChangeEpoch(_) => unsupported_tx_kind,
```

Comment:
match multi-values or simply keep them on the bottom and use `other` as the last matching arm
```suggestion
            SingleTransactionKind::TransferObject(_) | SingleTransactionKind::Publish(_) 
 | SingleTransactionKind::Call(_) | SingleTransactionKind::ChangeEpoch(_) => anyhow::bail!("Transaction kind not supported for estimator yet"),
```